### DISCUSSION TITLE
How to use panel-tabset for code and output of code.

### DISCUSSION THREAD

**Discussion by OffByOne37** [quarto-dev/quarto-cli#7326] [Q&A]
### Description

Hello,
I have a problem, that I want to show the plot of my code on my slide. However there should also be the possibility to show the code via a tabset. But I do not want to duplicate the code like this:

::: {.panel-tabset}

## Scatterplot
```{r}
#| fig-label: fig_try
#| fig-width: 5
#| fig-height: 3
#| echo: false
#| warning: false
#| out-width: 100%

#imagine code for plot generation here

```

## Code
```{r}
#| fig-label: fig_try
#| fig-width: 5
#| fig-height: 3
#| eval: false
#| echo: true
#| out-width: 100%

#imagine code for plot generation here

```
:::

Is there a way to use there a different way?

**Comment by mcanouil** [comment]
You don't need to go complex.
1. about reusing code via knitr: <https://bookdown.org/yihui/rmarkdown-cookbook/reuse-chunks.html>
2. it's `label` not `fig-label`.
3. `label` should not use underscore especially if you plan on using LaTeX/PDF.
4. No need for labels, unless you want to cross-reference the figures.
5. Quarto way of doing this,
   - see `panel` https://quarto.org/docs/reference/cells/cells-knitr.html#cell-output
   - See `output-location` https://quarto.org/docs/reference/cells/cells-knitr.html#cell-output

<table>
<tr><th>panel</th><th>output-location</th></tr>
<tr>
<td>

````qmd
---
format: revealjs
---

## Slide

```{r}
#| fig-width: 5
#| fig-height: 3
#| echo: false
#| warning: false
#| panel: tabset

plot(1)
```
````

</td>
<td>

````qmd
---
format: revealjs
---

## Slide

```{r}
#| fig-width: 5
#| fig-height: 3
#| echo: false
#| warning: false
#| output-location: column

plot(1)
```
````

</td>
</tr>
</table>
